### DISCUSSION TITLE
Control box design

### DISCUSSION THREAD

**Discussion by peterantypas** [peterantypas/maiana#17] [Ideas]
I'm going to leave this ticket open as a discussion forum around the control box. This is the unit on the opposite end of the transponder cable. It's not a strict requirement for the transponder to operate, but I have found mine useful over the past 2 years.

I started by designing custom cutouts in a Hammond enclosure and 3D printing it. I expect this process to take a few iterations over a couple of months.

![Screenshot from 2020-11-21 15-19-58](https://user-images.githubusercontent.com/1565933/99889826-a935bb80-2c0d-11eb-9e4f-b2b5a80609c9.png)

I don't intend to make this an IP6x rated enclosure. This unit is supposed to be inside the boat cabin or some other "mostly dry" environment.

The current design features an on/off latching switch, 4 status LEDs, a transmit on/off switch, and plain wire interfaces for 12V input and NMEA+/NMEA- outputs as well as USB.

**Comment by kagouraki** [comment]
Hello ! 
I am a cnc machinist, an electronics hobbyst and a sailor. 
I am very interested in your project ! 
If i can help with design and manufacture of the control box just tell me !

**Comment by peterantypas** [comment]
Thanks, I'll keep this in mind. Once I have the design down, maybe you can help me machine the cutouts in the stock enclosures. The stock box is very cheap (under $3), but Hammond wants quite a bit of money per unit to CNC those cutouts.

**Comment by kagouraki** [comment]
I will be glad to help ! Just say the word !

**Comment by peterantypas** [comment]
Some progress ... at least mechanical fit is almost there:

![IMG_0958](https://user-images.githubusercontent.com/1565933/102677901-a7eaa680-4159-11eb-8ccc-cef2df531620.jpg)
![IMG_0957](https://user-images.githubusercontent.com/1565933/102677902-aa4d0080-4159-11eb-8816-ae96f1899ae8.jpg)

**Reply by Corsair-63** [reply]
Nice design,
what I see are two switches, LAN interface and USB-C port, I guess the other should be power supply.

the device is it designed as one box or two boxes?
the useful one should be one box only, leisure vessel uses to be quite tight space and for devices even more.

so where are the GNSS aerial plug input and also the VHF plug?

I know I use to ask to many question, and of course that you should have everything planned but it's said than a pair of eyes more uses to see more than one pair ;-)

**Reply by peterantypas** [reply]
If you read the main project [README](https://github.com/peterantypas/maiana), it explains how MAIANA is different from everything else out there. In short, it is a VHF antenna with a built in transponder and GPS in its housing. There are no RF signals to and from the unit. What connects it to the rest of the world is a plain Cat5 cable which is not Ethernet, it just carries these 8 signals:

- 12V (Reverse polarity and ESD protected)
- GND (Reverse polarity and ESD protected)
- UART_RX (command input, 3.3V, buffered, ESD protected)
- UART_TX (NMEA output, 3.3V, buffered, ESD protected)
- TX_DISABLE (active high, buffered, ESD protected)
- GPS_STATUS (open drain, current limited, ESD protected)
- RX_STATUS (open drain, current limited, ESD protected)
- TX_STATUS (open drain, current limited, ESD protected)

As you can see, the unit is completely standalone. If you simply supply 12V/GND, it will just do its thing autonomously, according to the defaults. Now, that's not very useful, because you want at least the NMEA stream from it, hence the UART signals. That's where the control box comes in. But you could build your own if you were so inclined ...

This control box has a master on/off switch (red), a TX on/off switch (black), a microUSB connector for a serial -> USB interface via a CH330N (not sure if this has a Windows ARM driver, need to revisit), and a couple of stuffing options for either NMEA+/- (RS422) or plain UART signal to wire to something else. It also features short circuit protection, just in case. 

I took this approach because I didn't want to have yet another MCU in the control box, which is what I did with the earlier prototype on my boat. This isn't cast in stone, I could conceivably reduce the signals to 4 (12V, GND, RX, TX) and do the rest in the control box via additional logic, but I'm not crazy about that architecture. I could be swayed either way on this, though.

**Reply by Corsair-63** [reply]
It would be no problem to get the NMEA string from the USB, but the receiving computers will have to recognize it and in such case must be a COM port driver in order stablish connection transceiver > computer.
just in case that should not be possible could be the option of 3 connectors in the box for converting NMEA (COM) > USB (computer):
[https://digitalyacht.es/producto/adaptador-nmea-usb/](url) 
[https://actisense.com/products/opto-4-serial-opto-isolator/](url)
[https://actisense.com/products/usg-2-nmea-0183-converter/](url)
some examples about conversion.

otherwise PC could not read te NMEA stream received from the transceiver.

I cannot remember really well but Class B transponders have some fixed data, like MMSI, GNSS aerial position, name, and so on but another that "should" be exchanged, not sure about this last, I've been working always with Class A and data are defined as Static data (won't changes) and Voyage data as per its own name these exchanges every voyage, like crew on board, destination, draught, ... and also navigational status, has to be manually exchanged.

but maybe I'm mixing everything between A & B, that's the problem of having thousands of IMO Codes for sailing in your brain :-(

**Reply by peterantypas** [reply]
(Something is wrong with these links on github, I have to copy and paste the text to see them)

I think the first item will work, depending on the exact signals, but the NMEA+/- interface in the control box is output only. To make it bidirectional I would need to enlarge the design a bit, or dedicate the 4-signal Phoenix connector to NMEA0183, with a different cable for power. That might actually be a better way to go ...

**Reply by Corsair-63** [reply]
I'll leave it as text only, because also it redirects to me to a GitHub page.

https://digitalyacht.es/producto/adaptador-nmea-usb/
https://actisense.com/products/opto-4-serial-opto-isolator/
https://actisense.com/products/usg-2-nmea-0183-converter/

now they looks like are working

**Reply by peterantypas** [reply]
For reference, here is the actual size of the enclosure:

![IMG_0960](https://user-images.githubusercontent.com/1565933/102695278-0ad05200-41db-11eb-9c0a-36483d05c5eb.jpg)

**Reply by peterantypas** [reply]
For provisioning station data into class B, the status quo varies by legal jurisdiction. Here in the US, our nanny Government makes it illegal for you (the buyer) to do that on your own. That said, I don't exist for the purpose of pleasing the government. I intend to do to the AIS industry what Uber did to taxis. Regulations are designed to protect industry incumbents, not to advance innovation.

Long story short, you will need a serial interface into the unit to send commands for programming MMSI, name, call sign, etc. That's a one-time thing.

**Reply by Corsair-63** [reply]
Really nice and tiny enclosure, that should be marvellous to fir it on board :-)

**Reply by peterantypas** [reply]
That's the idea. MAIANA&trade; is designed for small boats. I'm hoping to offer a fully portable, battery powered, wireless version at some point, which would be absolutely ILLEGAL in the US, but like I said, I'm following the Uber model here ... that's the only way to innovate.

**Comment by jrcardwell** [comment]
Nice!

Jim

**Comment by peterantypas** [comment]
I installed the latest design (board 10.0.1) and the new control box on the boat yesterday. So far so good, pending some actual field testing (aka going sailing).

**Comment by peterantypas** [comment]
I just received a sample from a Chinese vendor of RJ45 connectors that seems a good fit and costs 10x less than the Wurth part I originally used. The main constraint here is that the RJ45 must be a mid-mount, low-profile type and those cost a lot more than conventional ones.

Will spin a new rev to confirm, but I'm optimistic this is going to be it for the control box design.
